sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/Acl.java

```diff
@@ -42,14 +42,34 @@
 
 
   /**
-   * The gRPC status code to respond with when access is denied.
+   * The status code to respond with when access is denied.
    *
-   * By default, this will be 7 (permission denied), but alternatives might include 16 (unauthenticated) or 5 (not
-   * found). If 0, indicates that the code should be inherited from the parent (regardless of the inherit field).
+   * By default, this will be 'Forbidden', but alternatives might include 'Authentication required' or 'Not
+   * Found'. If 'Inherited', indicates that the code should be inherited from the parent (regardless of the inherit field).
    *
-   * When HTTP transcoding is in use, this code will be translated to an equivalent HTTP status code.
    */
-  int denyCode() default 0;
+  DenyStatusCode denyCode() default  DenyStatusCode.FORBIDDEN_403;
+
+  enum DenyStatusCode {
+    INHERITED(0),
```

Comment:
There was an OK before but receiving an OK from a call that has been denied I find it counterintuitive for the user. Could you maybe elaborate?

Reply:
Oh yeah, it's counterintuitive for sure. I just think that since we are letting the user configure it, maybe we should provide that option as well. But it's also something we can easily add later so I'm happy to go with this version as well.

Reply:
What we probably can do, is to extract this enums to some other place. Not part of the ACL annotation. 

My reasoning is that with native components, calls to `effect.error("msg", code)` should receive our own http code. 

We should not allow users to pass raw integers, and we may want to offer a limited set of status codes to have better control of the conversion to gRPC codes. 

In addition to that, we may want to also offer (later) a limited set of 2xx codes. What I have in mind, is that a native code-first API may have effects that can be tweaked to returned different status codes (also for successful calls, ie: 2xx).

Reply:
I like that. One caveat I see. What do we do with `Acl.DenyStatusCode.INHERITED`, that only has meaning for the ACL? As per documentation  `If '0' [me: that is inherited], indicates that the code should be inherited from the parent (regardless of the inherit field).`
Maybe we could set the ACL default as `0`, INHERITED, and delete it from the list.

Reply:
Nope, I won't be able to set it to `0`. It must be an enum.

Reply:
Umm, that's indeed an issue.

Reply:
We have a long talk to discuss it. Current decision is to keep the status code inside the ACL annotation. As such we can ensure that only a subset of status code can be used for a deny code. 

And we plan to introduce a boolean (`inheritDenyCode`) to make it more explicit that the intention is to inherit from the parent. Also, because `0` maps to http status `Ok`.  

And we need to check how the proxy is dealing with `0` as well. No deny code, means default protobuf value, which is `0` for integer. We believe that the proxy is will ultimately convert `0` to `7` (permission denied) when it reaches the top parent ACL. However, that's just a guess.